modules/sql-engine/src/main/codegen/includes/parserImpls.ftl

```diff
@@ -532,3 +532,96 @@ SqlDrop SqlDropZone(Span s, boolean replace) :
         return new IgniteSqlDropZone(s.end(this), ifExists, zoneId);
     }
 }
+
+SqlNode SqlAlterZone() :
+{
+    final Span s;
+    final SqlIdentifier zoneId;
+    final SqlIdentifier newZoneId;
+    SqlNodeList optionList = null;
+}
+{
+    <ALTER> { s = span(); }
+    <ZONE> zoneId = CompoundIdentifier()
+    (
+      <RENAME> <TO> newZoneId = CompoundIdentifier() {
+        return new IgniteSqlAlterZoneRenameTo(s.end(this), zoneId, newZoneId);
+      }
+      |
+      <WITH> { s.add(this); } optionList = AlterZoneOptions() {
```

Comment:
IMO, we should use `WITH` keyword to provide context of the operation. To alter a certain property `SET` seems more natural to me. WDYT?

Reply:
Sure.